app/main/views/audits.py

```diff
@@ -83,23 +90,43 @@ def list_audits():
     if object_type:
         if object_type not in AUDIT_OBJECT_TYPES:
             abort(400, 'invalid object-type supplied')
-        if not object_id:
-            abort(400, 'object-type cannot be provided without object-id')
+
         model = AUDIT_OBJECT_TYPES[object_type]
         id_field = AUDIT_OBJECT_ID_FIELDS[object_type]
 
-        ref_object = model.query.filter(
-            id_field == object_id
-        ).first()
+        audits = audits.filter(AuditEvent.object.is_type(model))
 
-        if ref_object is None:
-            abort(404, "Object with given object-type and object-id doesn't exist")
+        if object_id:
+            ref_object = model.query.filter(
+                id_field == object_id
+            ).first()
 
-        audits = audits.filter(AuditEvent.object == ref_object)
+            if ref_object is None:
+                abort(404, "Object with given object-type and object-id doesn't exist")
 
+            audits = audits.filter(
+                AuditEvent.object_id == class_mapper(model).identity_key_from_instance(ref_object)[1][0]
```

Comment:
Yes, this is the only way I can see of checking the object type separately from the id. `.identity_key_from_instance(ref_object)[1][0]` indeed looks horrible but is exactly what `sqlalchemy_utils` does in its generic relationship code. So, if `sqlalchemy_utils` considers the API stable enough to depend on, I figured we may as well.

There was also a slightly less scary but more verbose way of doing it I discovered at the time which I could try.